pkg/cloud/services/networking/securitygroups.go

```diff
@@ -200,7 +202,29 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 	workerRules := append([]resolvedSecurityGroupRuleSpec{}, defaultRules...)
 
 	controlPlaneRules = append(controlPlaneRules, getSGControlPlaneHTTPS()...)
-	workerRules = append(workerRules, getSGWorkerNodePort()...)
+
+	// Fetch subnet to use for worker node port rules
+	// In the future IPv6 support need to be added here
+	if openStackCluster.Status.Network != nil && len(openStackCluster.Status.Network.Subnets) > 0 {
```

Comment:
Iterating over a nil or empty list is safe and idiomatic. The loop below performs this check implicitly anyway.

```suggestion
	if openStackCluster.Status.Network != nil {
```